applications/GeoMechanicsApplication/custom_processes/apply_constant_phreatic_surface_pressure_process.hpp

```diff
@@ -196,61 +139,20 @@ class ApplyConstantPhreaticSurfacePressureProcess : public Process
     double mEqRHS;
     double mPressureTensionCutOff;
 
-///----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-
 private:
 
     void calculateEquationParameters()
     {
-        Vector3 v1;
-        Vector3 v2;
-        v1 = mSecondReferenceCoordinate - mFirstReferenceCoordinate;
-        v2 = mThirdReferenceCoordinate - mFirstReferenceCoordinate;
-
-        mNormalVector[0] = v1[1] * v2[2] - v1[2] * v2[1];
-        mNormalVector[1] = v1[2] * v2[0] - v1[0] * v2[2];
-        mNormalVector[2] = v1[0] * v2[1] - v1[1] * v2[0];
-
-        double sum=0;
-        for (unsigned int i=0; i < mNormalVector.size(); ++i)
-        {
-            sum += mNormalVector[i]*mNormalVector[i];
-        }
-        double norm = std::sqrt(sum);
-        if (!(norm > 0.0))
+        const Vector3 v1 = mSecondReferenceCoordinate - mFirstReferenceCoordinate;
+        const Vector3 v2 = mThirdReferenceCoordinate  - mFirstReferenceCoordinate;
+        mNormalVector = MathUtils<double>::CrossProduct(v1, v2);
+        if (norm_2(mNormalVector) <= 0.0)
```

Comment:
```suggestion
        if (norm_2(mNormalVector) == 0.0)
```

(In the future we will need something better, e.g. a function `IsZeroVector`.)

Reply:
Done